TAOK3.py

```diff
@@ -0,0 +1,376 @@
+    If input contains one number the function accepts it as a maximum bound
+    Examples: gc_bounds=(20,40)
+              gc_bounds=40  # (0,40)
+
+    - length_bounds(tuple or int or float): contain mini and max length bounds
+    By default length_bounds=(0,4294967296)
+    If input contains one number the function accepts it as a maximum bound
+    Examples: length_bounds=(10,90)
+              length_bounds=90  # (0,90)
+
+    - quality_threshold(int or float): the quality threshold of the interest
+    By default quality_threshold=0
+    Examples: quality_threshold=10
+
+    There are three functions that are used in the main function:
+
+    - is_in_gc_bounds(seq_record, gc_bounds):
+    Check if the sequence falls in the range of GC-content bounds
+
+    - is_in_length_bounds(seq_record, length_bounds):
+    Check if the sequence falls in the range of length bounds
+
+    - is_above_quality_threshold(seq_record, quality_threshold):
+    Check if the mean of quality values exceeds the quality threshold
+
+    Return:
+    - file: file with fastq extension containing selected fragments.
+
+    For more information please see README
+
+    """
+    if type(length_bounds) is int:
+        length_bounds = 0, length_bounds
+    if type(gc_bounds) is int or type(gc_bounds) is float:
+        gc_bounds = 0, gc_bounds
+    sequences = SeqIO.parse(input_path, "fastq")
+    good_reads = []
+    for seq_record in sequences:
+        if (is_in_gc_bounds(seq_record, gc_bounds) and
+            is_in_length_bounds(seq_record, length_bounds) and
+            is_above_quality_threshold(seq_record, quality_threshold)):
+            good_reads += [seq_record]
+    if len(good_reads) == 0:
+        raise ValueError('There are no sequences suited to requirements')
+    if output_filename is None:
+        input_filename = os.path.split(input_path)[-1]
+        output_filename = input_filename
+    if not (output_filename.endswith('.fastq')):
+        output_filename = output_filename + '.fastq'
+    current_directory = os.getcwd()
+    path = os.path.join(current_directory, 'fastq_filtrator_results')
+    if not (os.path.exists(path)):
+        os.mkdir(path)
+    output_path = os.path.join(path, output_filename)
+    if os.path.exists(output_path):
+        error = 'File with such name exists! Change output_filename arg!'
+        raise ValueError(error)
+    SeqIO.write(good_reads, output_path, "fastq")
+
+
+class BiologicalSequence(ABC):
+    """
+    The abstract class for biological sequences
+    """
+    @abstractmethod
+    def __len__(self):
+        pass
+
+    @abstractmethod
+    def __getitem__(self, item):
+        pass
+
+    @abstractmethod
+    def __str__(self):
+        pass
+
+    def __repr__(self):
+        pass
+
+    @abstractmethod
+    def is_alphabet_correct(self):
+        pass
+
+
+class NucleicAcidSequence(BiologicalSequence):
+    """
+    Class for nucleic acids
+    """
+    def __init__(self, seq):
+        self.seq = seq
```

Comment:
Для стандартизации тут неплохо было бы сделать self.seq = seq.upper() (или .lower()) - не придется приводить к определенному регистру в будущем/сравнивать несколько регистров как на строчке 246
```suggestion
    def __init__(self, seq):
        self.seq = seq.upper()
```